src/handle.ml

```diff
@@ -77,6 +77,49 @@ let handle_tactic : sig_state -> Proof.t -> p_tactic -> Proof.t =
   | P_tac_print
   | P_tac_proofterm
   | P_tac_focus(_)      -> assert false (* Handled above. *)
+  | P_tac_assert(h,t)     ->
+     begin
+       (* Obtaining the goals environment, type and meta variable. *)
+       let env = fst (Proof.Goal.get_type g) in
+       let t = fst (Scope.scope_term StrMap.empty ss env t) in
+       let m = Proof.Goal.get_meta g in
+       (* Enriching the environnement with a var of type t *)
+       let h = match h with
+         | None   -> "h"
+         | Some x -> x.elt
+       in
+       let var = Bindlib.new_var Terms.mkfree h in
```

Comment:
I'm wondering if you took the simplest approach in the following (I did not really look at the details yet). What I would have done is call refine with a term of the form: `(λ(h:t),_) _`, where the first `_` corresponds to the previous goal with the extra hypothesis `h`, and the second `_` corresponds to the proof of `h`. I can explain more in person Monday if you want.